From: Takashi Iwai <tiwai@suse.de>
Subject: Add kABI placeholders for ASoC
Patch-mainline: Never, kABI padding
References: bsc#1179531

Add a few kABI padding fields to ASoC objects that are embedded in
other structs.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/soc-component.h |    2 ++
 include/sound/soc.h           |    6 ++++++
 2 files changed, 8 insertions(+)

--- a/include/sound/soc-component.h
+++ b/include/sound/soc-component.h
@@ -254,6 +254,8 @@ struct snd_soc_component {
 
 	struct dentry *debugfs_root;
 	const char *debugfs_prefix;
+
+	void *suse_kabi_padding;	/* XXX SLE-specific kABI placeholder */
 };
 
 #define for_each_component_dais(component, dai)\
--- a/include/sound/soc.h
+++ b/include/sound/soc.h
@@ -920,6 +920,8 @@ struct snd_soc_aux_dev {
 
 	/* codec/machine specific init - e.g. add machine controls */
 	int (*init)(struct snd_soc_component *component);
+
+	void *suse_kabi_padding;	/* XXX SLE-specific kABI placeholder */
 };
 
 /* SoC card */
@@ -1051,6 +1053,8 @@ struct snd_soc_card {
 	unsigned int component_chaining:1;
 
 	void *drvdata;
+
+	void *suse_kabi_padding;	/* XXX SLE-specific kABI placeholder */
 };
 #define for_each_card_prelinks(card, i, link)				\
 	for ((i) = 0;							\
@@ -1138,6 +1142,8 @@ struct snd_soc_pcm_runtime {
 
 	bool initialized;
 
+	void *suse_kabi_padding;	/* XXX SLE-specific kABI placeholder */
+
 	int num_components;
 	struct snd_soc_component *components[]; /* CPU/Codec/Platform */
 };
